activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -154,7 +154,9 @@ def setup_initial_database_yaml
         begin
           Rails.application.config.load_database_yaml
         rescue
-          $stderr.puts "Rails couldn't infer whether you are using multiple databases from your database.yml and can't generate the tasks for the non-primary databases. If you'd like to use this feature, please simplify your ERB."
+          unless Rails.application.config.active_record.suppress_multiple_database_warning
```

Comment:
This doesn't need to access `Rails`, we can get the value from Active Record directly.

```suggestion
          unless ActiveRecord::Base.suppress_multiple_database_warning
```